src/main/java/net/minestom/server/extensions/Extension.java

```diff
@@ -55,6 +65,79 @@ protected Logger getLogger() {
         return logger;
     }
 
+    public Path getDataDirectory() {
+        return dataDirectory;
+    }
+
+    /**
+     * Gets a resource from the extension directory, or from inside the jar if it does not
+     * exist in the extension directory.
+     * <p>
+     * If it does not exist in the extension directory, it will be copied from inside the jar.
+     *
+     * @param fileName The file to read
+     * @return The file contents, or null if there was an issue reading the file.
+     */
+    @Nullable
+    public InputStream getResource(@NotNull String fileName) {
```

Comment:
I wonder if this should be a path instead - they can be relative to the jar/data directory. I just hare dealing with separators :')